Docs (#1442)
IBAMR/IBAMR: doc/ci.md

for monitoring builds.

## How it works

In order to compile IBAMR we first need a basic computing environment (MPI,
compilers, a shell, etc.) as well as IBAMR's dependencies. To this end, all of
IBAMR's build dependencies are packaged inside one of several docker containers.
These containers provide a portable environment in which we can compile the
library with specified precompiled versions of PETSc, SAMRAI, etc. Most of our
containers are based on recent releases of Fedora: one is based on CENTOS 7 to
verify that the library still works with GCC 4.8 and minimal C++11 support.

The file specifying what GitHub does is `.github/workflows/push-pull.yml`. This
YAML file specifies what each virtual machine (hosted by GitHub) should do when
a new pull request is created. This file is responsible for setting up the
environment for compiling the library: in particular, it pulls down David Wells'
Docker images (either Fedora 33 or CENTOS 7) from DockerHub to get copies of all
recent dependencies.

Scripts for compiling docker images are available at

https://github.com/IBAMR/docker-files

## Improving performance with sccache and ccache

> The Docker setup we inherited from the Kitware guys used sccache: for all

Reviewer: ```suggestion
The Docker setup we use was initially developed by Kitware and uses sccache: for all
```
